test/framework/modules.py

```diff
@@ -168,18 +279,26 @@ def test_exist(self):
 
         java_mod_dir = os.path.join(self.test_prefix, 'Java')
         write_file(os.path.join(java_mod_dir, '1.8.0_181'), '#%Module')
+        write_file(os.path.join(self.test_prefix, 'toy', '42.1337'), '#%Module')
 
         if self.modtool.__class__ == EnvironmentModulesC:
             modulerc_tcl_txt = '\n'.join([
                 '#%Module',
                 'if {"Java/1.8" eq [module-info version Java/1.8]} {',
                 '    module-version Java/1.8.0_181 1.8',
                 '}',
+                'if {"Java/site_default" eq [module-info version Java/site_default]} {',
+                '    module-version Java/1.8.0_181 site_default',
+                '}',
             ])
         else:
             modulerc_tcl_txt = '\n'.join([
                 '#%Module',
                 'module-version Java/1.8.0_181 1.8',
+                'module-version Java/1.8.0_181 site_default',
+                'module-alias Java/Alias toy/42.1337',
+                # 'module-alias Java/NonExist non_existant/1',  # (only) LMod has this in module avail, disable for now
```

Comment:
Should still this be commented out?

You're running a test with `Java/NonExist` below, so it shouldn't be commented?

You can only test whether `Java/NonExist` is included in the output of `avail` when `Lmod` is being used?

Reply:
> You're running a test with Java/NonExist below, so it shouldn't be commented?

Yes, with expected "False", so this also test if no other alias is used.

And no, LMod will always fail the test as exists checks avail first returning True although using "show" would return False. Hence my (now strong) suggestion to remove the exists-via-avail code